### PR TITLE
API-66:  docs for promotions API

### PR DESCRIPTION
**Pull request by laher**
I have documented endpoints and associated types.

# Documented endpoints:
 * List promotions
 * Create a promotion
 * Get a single promotion
 * Update promotion
 * Get products for a promotion
 * Get promotion promo codes
 * Apply promotion to a sale (`/discount`)
   * NOTE: `/discount` would normally be applied internally as part of a sale request, and it's doubtful that it'd be useful in its current form. Including it here for the purposes of a conversation with Marsello, but it may well be removed before merging this PR.

## Excluded (open to discussion):
 * Preview a promotion (used in the Vend app while editing a promotion)
 * Bulk promo code status / delete (unused by the vend app; seems unnecessary because 'Update' can do this)
 * 'internal' endpoints

NOTE: I haven't added examples yet, but still ...

irrefutable

### CODE REVIEW

**Review by pzurek — APPROVED** [review]
LGTM 🚢 🚢 🚢

File: api.json [hunk 1]
```diff
@@ -4489,6 +4798,219 @@
...
+          "format": "double"
+        },
+        "promotion_ids": {
+          "description": "List of promotion IDs applied to this line item.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "metadata": {
```
Comment (Wal8800) [inline api.json]: We don't actually use `metadata` field for line items so not sure if it's worth mentioning in the doc? 🤔

**Review by Wal8800 — APPROVED** [review]
LGTM, one minor question

File: api.json (hunk 1 above)
Comment (laher) [reply api.json]: Hmm, in that case I might take it out completely.

It's either that or put a big 'UNUSED' label across it.

Comment (pzurek) [reply api.json]: I'd say better to take it out and avoid any confusion or give people something to ask about.

**Pull request merged**